tools/undo-block/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+use chrono::Utc;
+use near_chain::types::LatestKnown;
+use near_chain::{ChainStore, ChainStoreAccess, ChainStoreUpdate};
+use near_chain_configs::GenesisValidationMode;
+use near_primitives::block::Tip;
+use near_primitives::utils::to_timestamp;
+use near_store::{Mode, NodeStorage};
+use nearcore::load_config;
+use std::path::Path;
+
+#[derive(clap::Parser)]
+pub struct UndoBlockCommand {}
+
+impl UndoBlockCommand {
+    pub fn run(
+        self,
+        home_dir: &Path,
+        genesis_validation: GenesisValidationMode,
+    ) -> anyhow::Result<()> {
+        let near_config = load_config(home_dir, genesis_validation)
+            .unwrap_or_else(|e| panic!("Error loading config: {:#}", e));
+
+        let store_opener = NodeStorage::opener(
+            home_dir,
+            near_config.config.archive,
+            &near_config.config.store,
+            None,
+        );
+
+        let storage = store_opener.open_in_mode(Mode::ReadWrite).unwrap();
+        let store = storage.get_hot_store();
+
+        let mut chain_store = ChainStore::new(
+            store,
+            near_config.genesis.config.genesis_height,
+            near_config.client_config.save_trie_changes,
+        );
+
+        let current_head = chain_store.head()?;
+        let current_head_hash = current_head.last_block_hash;
+        let prev_block_hash = current_head.prev_block_hash;
+        let prev_header = chain_store.get_block_header(&prev_block_hash)?;
+        let prev_tip = Tip::from_header(&prev_header);
+        let current_head_height = current_head.height;
+        let prev_block_height = prev_tip.height;
+        
+        tracing::info!(target: "neard", ?prev_block_hash, ?current_head_hash, "Trying to update head");
+        tracing::info!(target: "neard", ?prev_block_height, ?current_head_height, "Trying to update head");
+
+        // stop if it's already the final block
+        if chain_store.final_head()?.height >= current_head.height {
```

Comment:
this tool only removes head block and set head to head's prev block. This tool is unable to undo a random block in the middle of a chain.

Reply:
Oh, sure! But then it might be cleaner to rename `clear_block_data_undo_block` to `clear_chain_head` and retrieve the chain head within the method instead of hash comparison, as it seems to be specifically designed for that purpose.